packages/gateway/src/resolvers/domain.ts

```diff
@@ -0,0 +1,62 @@
+import { Hex, labelhash, namehash, zeroAddress } from 'viem'
+
+import { Text, Address, DomainMetadata, NodeProps } from '../types'
+import { Domain } from '../entities'
+
+interface ReadRepository {
+  getDomain(params: NodeProps): Promise<Domain | null>
+  getSubdomains({ node }: NodeProps): Promise<string[]>
+  getTexts({ node }: NodeProps): Promise<Text[]>
+  getAddresses({ node }: NodeProps): Promise<Address[]>
+}
+
+interface Client {
+  getOwner(node: Hex): Promise<Hex>
+  getResolver(node: Hex): Promise<Hex | undefined>
+}
+
+export async function domainResolver(
+  { name }: { name: string },
+  repo: ReadRepository,
+  client: Client,
+): Promise<DomainMetadata | undefined> {
+  const node = namehash(name)
```

Comment:
Name should be normalized before calculating node.

Please create an issue for checking this in the whole codebase. It should be in **only** on **reading**, not on writing. (That's also a question to do, please add it in our notes)